fix(web): fix #4574 link button + improves accessibility

BogdanCerovac opened the pull request:
Button in a link fixed,
horizontal rule is decorative here 
and `aria-current="page"` informs screen-readers that user is on that page when focusing on the link.

fixes #4574

BogdanCerovac committed:
Fixed semantic problem in navigation-bar.svelte

Closes [4574]

BogdanCerovac committed:
Reintroduced styling for navigation-bar.svelte

Based on button styling

BogdanCerovac committed:
Horizontal rule set as decoration in navigation-bar.svelte

BogdanCerovac committed:
aria-current added as an assistive technology indication for current page

Horizontal rule indicates current page for users that can see the screen and with aria-current screen-reader users get the same information

jrasm91 commented:
Can you add to the description before/after screenshots if applicable?

BogdanCerovac committed:
Temporary fix for accessibility name of the link when SVG replaces text

This is a temporary fix as we first need to fix the svelte-material-icons to support role="img" on rendered SVGs.

BogdanCerovac commented:
@jrasm91 - sure, please check them below. I also had to make a temporary fix for accessibility name of the link with `aria-label` because it seems that svelte-material-icons don't have a prop to set proper role to it (should be `role="img"` (will make a PR there before we can have a permanent fix here)).


Screenshots before:

Desktop:
![Desktop - button nested inside a link](https://github.com/immich-app/immich/assets/8626513/b83ebd60-97a7-4adf-8633-07a1b7d22df3)

Mobile:
![Mobile - button nested inside a link, only icon shown](https://github.com/immich-app/immich/assets/8626513/56119a48-11e1-40f5-893f-8d09422976ec)

Screenshot after:

Desktop:
![Desktop - link without button](https://github.com/immich-app/immich/assets/8626513/b50ce957-3a48-40d1-bb2f-092039f48ff1)

Mobile:
![Mobile - link without button, only icon shown](https://github.com/immich-app/immich/assets/8626513/cf623cef-af06-473d-92eb-58e6bc52fc68)

BogdanCerovac commented:
Opened issue for SVG missing role in https://github.com/ramiroaisen/svelte-material-icons/issues/20

jrasm91 commented on web/src/lib/components/shared-components/navigation-bar/navigation-bar.svelte:
```diff
@@ -76,33 +76,38 @@
+                      : ''}"
+                  />
+                <hr
+                  role="decoration"
```
The linting checks are complaining that a hr cannot have a role. Can you fix/address this please?

BogdanCerovac replied:
Yes, I have a better solution on it's way.

BogdanCerovac committed:
fix(web): horizontal rule replaced with div

hr is not semantically correct, therefore we tried with role="decoration" (that should be role="presentation") but it's actually better to just use a div as it's best practice to not override semantics when we can avoid that...

Btw. the semantics for active element for assistive technology is added with previous commit setting aria-current on the link...

jrasm91 committed:
chore: format

jrasm91 commented:
I justed wanted to follow up on this - we don't use svelte-material-icons anymore and our own `Icon` component should support the `role` attribute now, which defaults to `role="img"`. Does that solve resolve the comment [here](https://github.com/immich-app/immich/pull/4575#issuecomment-1773902034)

BogdanCerovac commented:
@jrasm91 - yes, thank you. `role="img"` is perfect for SVGs used as images.

Now we must make sure that icons that bear information or are used instead of text in interactive elements also have their accessibility name set (as discussed in https://github.com/ramiroaisen/svelte-material-icons/issues/20#issuecomment-1779221649).

Will check and open an issue or PR...
